src/testdrive.F90

```diff
@@ -405,6 +539,34 @@ recursive subroutine run_unittest(test, unit, stat)
     !$omp critical(testdrive_testsuite)
     write(unit, '(a)') message
     !$omp end critical(testdrive_testsuite)
+    
+    stdout = ''
+    if (allocated(error)) then
+      stdout = trim(error%message)
+    endif
+    
+    s = index(message, '[')
+    e = index(message, ']')
+    res = message(s+1:e-1)
+    select case (res)
```

Comment:
Does this mean that this `select case` also happen, even if the user does not want JUnit.xml file?

Reply:
At the moment the JUnit.xml is written always.

Reply:
Up to @awvwgk to decide the final behaviour, but my preference would be htat JUnit.xml is written only on request of the user.

Reply:
My suggestion is that calling `junitxml_open_file()` switches a global variable from `.false.` to `.true`. If `.true` the output file is written. This requires that any of the `junitxml_write_*()` routines needs to respect this flag.
Let me know, if you propose a different way.
If not, I will implement as suggested.
Please, let me know your thoughts.

Reply:
That is an option.
Another possible option is to store the content in an `optional` `character` variable, and to write this variable at the end of all tests.
The `optional` variable would allow backward compatibility.

E.g.,
```fortran
character(len=:), allocatable :: xml_
        write(error_unit, fmt) "Suite:", testsuites(is)%name
        call run_selected(testsuites(is)%collect, test_name, error_unit, stat, xml =  xml_)
        call junitxml_write_testsuite(xml_)
```